test/OpenFeature.Tests/Telemetry/EvaluationEventBuilderTests.cs

```diff
@@ -0,0 +1,131 @@
+
+namespace OpenFeature.Tests.Telemetry
+{
+    public class EvaluationEventBuilderTests
+    {
+        [Fact]
+        public void Build_ShouldReturnEventWithCorrectAttributes()
+        {
+            // Arrange
+            var clientMetadata = new ClientMetadata("client", "1.0.0");
+            var providerMetadata = new Metadata("provider");
+            var hookContext = new HookContext<Value>("flagKey", new Value(), FlagValueType.Object, clientMetadata,
+                providerMetadata, EvaluationContext.Empty);
+            var metadata = new Dictionary<string, object>
+            {
+                { "flagSetId", "flagSetId" }, { "contextId", "contextId" }, { "version", "version" }
+            };
+            var flagMetadata = new ImmutableMetadata(metadata);
+            var details = new FlagEvaluationDetails<Value>("flagKey", new Value("value"), ErrorType.None,
+                reason: "reason", variant: "variant", flagMetadata: flagMetadata);
+
+            // Act
+            var evaluationEvent = EvaluationEventBuilder.Build(hookContext, details);
+
+            // Assert
+            Assert.Equal("feature_flag.evaluation", evaluationEvent.Name);
+            Assert.Equal("flagKey", evaluationEvent.Attributes[TelemetryConstants.Key]);
+            Assert.Equal("provider", evaluationEvent.Attributes[TelemetryConstants.Provider]);
+            Assert.Equal("reason", evaluationEvent.Attributes[TelemetryConstants.Reason]);
+            Assert.Equal("variant", evaluationEvent.Attributes[TelemetryConstants.Variant]);
+            Assert.Equal("contextId", evaluationEvent.Attributes[TelemetryFlagMetadata.ContextId]);
+            Assert.Equal("flagSetId", evaluationEvent.Attributes[TelemetryFlagMetadata.FlagSetId]);
+            Assert.Equal("version", evaluationEvent.Attributes[TelemetryFlagMetadata.Version]);
+        }
+
+        [Fact]
+        public void Build_ShouldHandleErrorDetails()
+        {
+            // Arrange
+            var clientMetadata = new ClientMetadata("client", "1.0.0");
+            var providerMetadata = new Metadata("provider");
+            var hookContext = new HookContext<Value>("flagKey", new Value(), FlagValueType.Object, clientMetadata,
+                providerMetadata, EvaluationContext.Empty);
+            var metadata = new Dictionary<string, object>
+            {
+                { "flagSetId", "flagSetId" }, { "contextId", "contextId" }, { "version", "version" }
+            };
+            var flagMetadata = new ImmutableMetadata(metadata);
+            var details = new FlagEvaluationDetails<Value>("flagKey", new Value("value"), ErrorType.General,
+                errorMessage: "errorMessage", reason: "reason", variant: "variant", flagMetadata: flagMetadata);
+
+            // Act
+            var evaluationEvent = EvaluationEventBuilder.Build(hookContext, details);
+
+            // Assert
+            Assert.Equal("General", evaluationEvent.Attributes[TelemetryConstants.ErrorCode]);
```

Comment:
This should be lowercase.

Reply:
Thanks for reminding me. Should be fixed with https://github.com/open-feature/dotnet-sdk/pull/397/commits/0d1a869424d98202f6f9ac5d76d0e7609e732967